### DISCUSSION TITLE
Global interceptor

### DISCUSSION THREAD

**Discussion by LonguCodes** [TypedProject/ts-express-decorators#2887] [Q&A] [answered]
Hi

I'm using `typescript-functiona-extensions` in my project and wanted to write a global middleware/inteceptor that would automatically unwrap the results of the controller. 

How can I achieve that?

Interceptor has to be applied per-controller
Middleware gets the response, not the returned object.

**Comment by Romakita** [comment]
Hi @LonguCodes 
You can use the $onResponse to handle $ctx and retrieve all request/response information.

look this page: https://rc.tsed.dev/docs/hooks.html

**Reply by LonguCodes** [reply]
I tried using the `$onResponse` hook, but changing `$ctx.data` does not change the response

**Reply by LonguCodes** [reply]
I see there is a line that may do what I need, but it's commented out from the beginning 
https://github.com/tsedio/tsed/blame/0158fb6d335ed76c653fb9797be9d41bc52b1be4/packages/platform/common/src/services/PlatformResponse.ts#L386

**Reply by Romakita** [reply]
Normal, this is not the good place to change the data. it's too late. $onResponse, is here to log data. Also, I haven't said, you need to modify response.data.

**Reply by LonguCodes** [reply]
I'm not saying $onResponse has some faults, just that it does not solve my issue, as it does not allow me to change the controller result.

**Reply by Romakita** [reply]
the commented code has never been used. it is an attempt at evolution that has created regressions. 
The hook is emitted here: https://github.com/tsedio/tsed/blob/19ee58b7c0738b2c33ee04f0845e89abbf3fe185/packages/platform/platform-http/src/common/domain/PlatformContext.ts#L88

**Reply by Romakita** [reply, marked as answer]
So have you looked on Response Filters feature, this is in the doc...

**Reply by LonguCodes** [reply]
Thank you, this worked. Didn't think ResponseFIlter would help due to the name. 
Thanks again!

**Reply by Romakita** [reply]
;)
